clang/test/SemaObjCXX/attr-nonblocking-constraints.mm

```diff
@@ -0,0 +1,19 @@
+// RUN: %clang_cc1 -fsyntax-only -fblocks -fcxx-exceptions -verify %s
+
+#pragma clang diagnostic ignored "-Wperf-constraint-implies-noexcept"
+
+// Objective-C
+@interface OCClass
+- (void)method;
+@end
+
+void nl14(OCClass *oc) [[clang::nonblocking]] {
+	[oc method]; // expected-warning {{'nonblocking' function must not access an ObjC method or property}}
+}
```

Comment:
I think it should also forbid `super` for the same reason: accessing it would mean going through a message

Reply:
Yeah... This gets into territory I hadn't ventured into yet, since `super` can't appear outside of the context of an ObjC method (right?), and currently, `nonblocking` and `nonallocating` are ignored on ObjC methods. The rationale was that you have to bend over backwards to bypass the ObjC runtime and call a method safely, and once you're in a method, you can't do much at all safely, so is it really worth the trouble?

So maybe the first question is how to fix this bug -- the following generates no diagnostics:

```
@implementation Derived
- (void)problem __attribute((nonblocking)) {
	throw 42;
}
@end
```

The easiest solution is to forbid the attribute in this context.

Reply:
> since super can't appear outside of the context of an ObjC method (right?)

correct

> The easiest solution is to forbid the attribute in this context.

sounds good

Reply:
This looks like a bit of a job of its own so I wrote
https://github.com/llvm/llvm-project/issues/100676